plugins/woocommerce/includes/admin/meta-boxes/views/html-order-item.php

```diff
@@ -10,7 +10,7 @@
 defined( 'ABSPATH' ) || exit;
 
 $product      = $item->get_product();
-$product_link = $product ? admin_url( 'post.php?post=' . $item->get_product_id() . '&action=edit' ) : '';
+$product_link = $product ? apply_filters( 'woocommerce_admin_order_item_product_link', admin_url( 'post.php?post=' . $item->get_product_id() . '&action=edit' ), $item_id, $item ) : '';
```

Comment:
It might be better to move this to the next line and run the filter on the $product_link variable. That way it can be filtered even if $product is a falsy value.

```suggestion
$product_link = $product ? admin_url( 'post.php?post=' . $item->get_product_id() . '&action=edit' ) : '';
$product_link = apply_filters( 'woocommerce_admin_order_item_product_link_meta_box', $product_link, $product, $item_id, $item );
```

Reply:
Also please add the filter documentation and @since tag.